metadata/service/exporter/configurable/exporter.go

```diff
@@ -46,12 +46,21 @@ func NewMetadataServiceExporter(metadataService service.MetadataService) exporte
 }
 
 // Export will export the metadataService
-func (exporter *MetadataServiceExporter) Export() error {
+func (exporter *MetadataServiceExporter) Export(url *common.URL) error {
 	if !exporter.IsExported() {
 		serviceConfig := config.NewServiceConfig(constant.SIMPLE_METADATA_SERVICE_NAME, context.Background())
 		serviceConfig.Protocol = constant.DEFAULT_PROTOCOL
-		serviceConfig.Protocols = map[string]*config.ProtocolConfig{
-			constant.DEFAULT_PROTOCOL: generateMetadataProtocol(),
+		if url != nil && url.SubURL != nil {
+			serviceConfig.Protocols = map[string]*config.ProtocolConfig{
+				constant.DEFAULT_PROTOCOL: {
+					Name: url.SubURL.Protocol,
+					Port: url.SubURL.Port,
+				},
+			}
+		} else {
```

Comment:
this else is not necessary

Reply:
When the URL is nil, it returns an error?

Reply:
Yes

Reply:
yes